assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js

```diff
@@ -0,0 +1,275 @@
+		select( CORE_NOTIFICATIONS ).isNotificationDismissalFinal( id )
+	);
+
+	// See TODO note below.
+	const isCTADismissed = useSelect( ( select ) =>
+		select( CORE_NOTIFICATIONS ).isNotificationDismissed( id )
+	);
+	const dismissedPromptsLoaded = useSelect( ( select ) =>
+		select( CORE_USER ).hasFinishedResolution( 'getDismissedPrompts', [] )
+	);
+
+	const autoSubmit = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue(
+			AUDIENCE_SEGMENTATION_SETUP_FORM,
+			'autoSubmit'
+		)
+	);
+
+	const [ showErrorModal, setShowErrorModal ] = useState( false );
+
+	const { dismissItem, dismissPrompt } = useDispatch( CORE_USER );
+
+	const { apiErrors, failedAudiences, isSaving, onEnableGroups } =
+		useEnableAudienceGroup( {
+			onSuccess: () => {
+				invalidateResolution( 'getQueuedNotifications', [
+					viewContext,
+					NOTIFICATION_GROUPS.SETUP_CTAS,
+				] );
+				dismissPrompt( id, {
+					expiresInSeconds: 0,
+				} );
+				// Dismiss success notification in settings.
+				dismissItem(
+					SETTINGS_VISITOR_GROUPS_SETUP_SUCCESS_NOTIFICATION
+				);
+			},
+			onError: () => {
+				setShowErrorModal( true );
+			},
+		} );
+
+	const { clearPermissionScopeError } = useDispatch( CORE_USER );
+	const { setSetupErrorCode } = useDispatch( CORE_SITE );
+
+	const onCancel = useCallback( () => {
+		setValues( AUDIENCE_SEGMENTATION_SETUP_FORM, {
+			autoSubmit: false,
+		} );
+		clearPermissionScopeError();
+		setSetupErrorCode( null );
+		setShowErrorModal( false );
+	}, [ clearPermissionScopeError, setSetupErrorCode, setValues ] );
+
+	const setupErrorCode = useSelect( ( select ) =>
+		select( CORE_SITE ).getSetupErrorCode()
+	);
+
+	const hasOAuthError = autoSubmit && setupErrorCode === 'access_denied';
+
+	useEffect( () => {
+		if ( isTooltipVisible ) {
+			trackEvent( trackEventCategory, 'tooltip_view' );
+		}
+	}, [ isTooltipVisible, trackEventCategory ] );
+
+	if ( isTooltipVisible ) {
+		return (
+			<Fragment>
+				<AdminMenuTooltip
+					title={ __(
+						'You can always enable groups from Settings later',
+						'google-site-kit'
+					) }
+					content={ __(
+						'The visitors group section will be added to your dashboard once you set it up.',
+						'google-site-kit'
+					) }
+					dismissLabel={ __( 'Got it', 'google-site-kit' ) }
+					onDismiss={ () => {
+						trackEvent( trackEventCategory, 'tooltip_dismiss' );
+					} }
+					tooltipStateKey={ id }
+				/>
+			</Fragment>
+		);
+	}
+
+	const hideCTABanner =
+		( isCTADismissed || ! dismissedPromptsLoaded ) && ! isSaving;
+
+	// TODO Remove this hack
+	// We "incorrectly" pass true to the `skipHidingFromQueue` option when dismissing this banner.
+	// This is because we don't want the component removed from the DOM as we have to still render
+	// the `AdminMenuTooltip` in this component. This means that we have to rely on manually
+	// checking for the dismissal state here.
+	if ( hideCTABanner && ! hasOAuthError && ! showErrorModal ) {
+		return null;
+	}
```

Comment:
```suggestion
	// TODO: Don't use `skipHidingFromQueue` and remove the need to check
	// if this component should output anything.
	//
	// We "incorrectly" pass true to the `skipHidingFromQueue` option when dismissing this banner.
	// This is because we don't want the component removed from the DOM as we have to still render
	// the `AdminMenuTooltip` in this component. This means that we have to rely on manually
	// checking for the dismissal state here.
	if ( hideCTABanner && ! hasOAuthError && ! showErrorModal ) {
		return null;
	}
```

Can you file an issue for this so we have a plan to remove it? TODOs like this in the code without an attached issue might get ignored, but if you file a follow-up we can reference it here.

Also: the note here doesn't quite seem right, as we _aren't_ rendering anything in the DOM when we `return null`. I assume you mean that we don't want the component removed from the component tree/React rendering, but not the DOM, right? 🤔

Reply:
I updated the comment, the issue is already created by @jimmymadon since this same approach is shared between few components